jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/ProcessDefinitionDataSetProviderTest.java

```diff
@@ -0,0 +1,118 @@
+import org.dashbuilder.dataset.DataSetLookup;
+import org.dashbuilder.dataset.DataSetMetadata;
+import org.dashbuilder.dataset.def.DataSetDef;
+import org.dashbuilder.dataset.filter.CoreFunctionFilter;
+import org.dashbuilder.dataset.filter.CoreFunctionType;
+import org.dashbuilder.dataset.filter.DataSetFilter;
+import org.jbpm.workbench.ks.integration.ConsoleDataSetLookup;
+import org.jbpm.workbench.pr.model.ProcessSummary;
+import org.jbpm.workbench.pr.service.ProcessRuntimeDataService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.COL_DYNAMIC;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.COL_ID_PROCESSDEF;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.COL_ID_PROCESSNAME;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.COL_ID_PROCESSVERSION;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.COL_ID_PROJECT;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.PROCESS_DEFINITION_DATASET;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProcessDefinitionDataSetProviderTest {
+
+    @Mock
+    protected ProcessRuntimeDataService processRuntimeDataService;
+
+    @InjectMocks
+    ProcessDefinitionDataSetProvider processDefinitionDataSetProvider;
+
+    private DataSetDef dataSetdef;
+
+    @Before
+    public void init() {
+        ProcessSummary processSummaryOne = new ProcessSummary("test-def-id", "test-def-name", "testone", "1", false);
+        ProcessSummary processSummaryTwo = new ProcessSummary("test-def-id", "test-def-name", "testtwo", "1", false);
+        ProcessSummary processSummaryThree = new ProcessSummary("test-def-id", "test-def-name", "testtwo", "2", false);
+
+        List<ProcessSummary> processSummaries = Arrays.asList(processSummaryOne, processSummaryTwo, processSummaryThree);
+        doAnswer(answer -> {
+            return processSummaries;
+        }).when(processRuntimeDataService).getProcessesByFilter(anyString(),
+                                                                anyString(),
+                                                                anyInt(),
+                                                                anyInt(),
+                                                                anyString(),
+                                                                anyBoolean());
+
+        dataSetdef = ProcessDefinitionDataSetDefBuilder.get()
```

Comment:
seems to be duplicating what is done in ProcessDefinitonDataSetGenerator

Reply:
I just want mock a datasetdef for my test, it is different to ProcessDefinitonDataSetGenerator.